### PR TITLE
6) Migrate anchor withdraw single token endpoint

### PR DESCRIPTION
**Pull request by elliotkennedy**
1) https://github.com/hubbleprotocol/hyperplane/pull/33 
2) https://github.com/hubbleprotocol/hyperplane/pull/36 
3) https://github.com/hubbleprotocol/hyperplane/pull/37 
4) https://github.com/hubbleprotocol/hyperplane/pull/38 
5) https://github.com/hubbleprotocol/hyperplane/pull/39 
6) https://github.com/hubbleprotocol/hyperplane/pull/40 <-- here

### CODE REVIEW

**Review by y2kappa — COMMENTED** [review]
lgtm

File: programs/hyperplane/src/instructions/withdraw_single_token_type.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    pub pool: AccountLoader<'info, SwapPool>,
+
+    /// CHECK: has_one constraint on the pool
+    pub swap_curve: UncheckedAccount<'info>,
+
+    /// CHECK: has_one constraint on the pool
+    pub pool_authority: AccountInfo<'info>,
+
+    /// CHECK: checked in the handler
+    pub destination_token_mint: Box<MultiProgramCompatibleAccount<'info, Mint>>,
```
Comment (andreihrs) [inline programs/hyperplane/src/instructions/withdraw_single_token_type.rs:143]: I would separate the pool accounts from the user specific accounts

File: programs/hyperplane/src/instructions/withdraw_single_token_type.rs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+    msg!(
+        "Swap pool inputs: swap_type={:?}, token_a_balance={}, token_b_balance={}, pool_token_supply={}",
+        swap_curve.curve_type,
+        ctx.accounts.token_a_vault.amount,
+        ctx.accounts.token_b_vault.amount,
+        ctx.accounts.pool_token_mint.supply,
+    );
+
+    let pool_mint_supply = to_u128(ctx.accounts.pool_token_mint.supply)?;
```
Comment (andreihrs) [inline programs/hyperplane/src/instructions/withdraw_single_token_type.rs:38]: maybe we can move these into an operations folder?

**Review by andreihrs — APPROVED** [review]

File: programs/hyperplane/src/instructions/withdraw_single_token_type.rs (hunk 2 above)
Comment (elliotkennedy) [reply programs/hyperplane/src/instructions/withdraw_single_token_type.rs:38]: Yeah I will consider refactoring - just copying into handlers for now

**Pull request merged**